File: receipt-analyzer-service/app/src/main/java/pl/madzierski/daniel/receipt/model/GetReceiptListResponse.java
package pl.madzierski.daniel.receipt.model;

import java.time.LocalDateTime;
import java.util.List;

public record GetReceiptListResponse(
    List<GetReceiptListItemResponse> items
) {
    public record GetReceiptListItemResponse(
        String id,
        String name,
        String description,
        LocalDateTime createDate
    ) {
    }
}

File: receipt-analyzer-service/src/main/kotlin/pl/madzierski/daniel/app/receipt/model/GetReceiptListResponse.kt
package pl.madzierski.daniel.app.receipt.model

import java.util.*

data class GetReceiptListResponse(
    val items: List<GetReceiptListItemResponse>
){
    data class GetReceiptListItemResponse(
        val id: String,
        val name: String?,
        val description: String?,
        val createDate: Date?,
    )
}